network/tests/test_linux.py

```diff
@@ -0,0 +1,338 @@
+            return decode_string(contents), None, None
+
+
+def read_int_file_mock(location):
+    if location == '/sys/class/net/lo/mtu':
+        return 65536
+    elif location == '/sys/class/net/lo/tx_queue_len':
+        return 1000
+    elif location == '/sys/class/net/ens5/mtu':
+        return 9001
+    elif location == '/sys/class/net/ens5/tx_queue_len':
+        return 1000
+    elif location == '/sys/class/net/invalid/mtu':
+        return None
+    elif location == '/sys/class/net/invalid/tx_queue_len':
+        return None
+
+
+@mock.patch('datadog_checks.network.network.Platform.is_linux', return_value=True)
+def test_returns_the_right_instance(is_linux, check):
+    check_instance = check({})
+    assert isinstance(check_instance, LinuxNetwork)
+
+
+@pytest.mark.skipif(not Platform.is_linux(), reason="Only works on Linux systems")
+def test_collect_cx_queues(check, aggregator):
+    instance = copy.deepcopy(common.INSTANCE)
+    instance['collect_connection_queues'] = True
+    instance['collect_connection_state'] = True
+    check_instance = LinuxNetwork('network', {}, [instance])
+
+    check_instance.check({})
+
+    for metric in CONNECTION_QUEUES_METRICS + common.EXPECTED_METRICS:
+        aggregator.assert_metric(metric)
+
+
+@pytest.mark.skipif(not Platform.is_linux(), reason="Only works on Linux systems")
+def test_collect_cx_queues_when_ss_fails(check, aggregator):
+    instance = copy.deepcopy(common.INSTANCE)
+    instance['collect_connection_queues'] = True
+    instance['collect_connection_state'] = True
+    check_instance = LinuxNetwork('network', {}, [instance])
+    with mock.patch('datadog_checks.network.check_linux.get_subprocess_output') as out:
+        out.side_effect = ss_subprocess_mock_fails
+        check_instance.check({})
+
+    for metric in CONNECTION_QUEUES_METRICS + common.EXPECTED_METRICS:
+        aggregator.assert_metric(metric)
+
+
+@pytest.mark.skipif(Platform.is_windows(), reason="Only runs on Unix systems")
+def test_cx_state(aggregator):
+    instance = copy.deepcopy(common.INSTANCE)
+    instance['collect_connection_state'] = True
+    check_instance = LinuxNetwork('network', {}, [instance])
+
+    with mock.patch('datadog_checks.network.check_linux.get_subprocess_output') as out:
+        out.side_effect = ss_subprocess_mock
+        check_instance.check(instance)
+        for metric, value in iteritems(CX_STATE_GAUGES_VALUES):
+            aggregator.assert_metric(metric, value=value)
+        aggregator.reset()
+
+        out.side_effect = netstat_subprocess_mock
+        check_instance.check(instance)
+        for metric, value in iteritems(CX_STATE_GAUGES_VALUES):
+            aggregator.assert_metric(metric, value=value)
+
+
+@pytest.mark.skipif(Platform.is_windows(), reason="Only runs on Unix systems")
+@mock.patch('os.listdir', side_effect=os_list_dir_mock)
+@mock.patch('datadog_checks.network.check_linux.LinuxNetwork._read_int_file', side_effect=read_int_file_mock)
```

Comment:
What is the team's general stance on using `mock.patch`? I personally try to only use it for legacy stuff, for new code I feel like I've designed something wrong if I have to mock deep inside a function instead of parametrizing (eg with dependency injection).

Reply:
We do not use dependency injection. In general I agree it is usually a smell if you need to use it although one might argue that some times extracting something to a separate module for the sake of correctness can be too much if that something is small and not shared with other things. I don't want to get too philosophical here but here all I'm doing is moving the tests from one file to another and change the SUT from a Network object to a LinuxNetwork object so I do not need to mock the Platform. This is very much legacy code still, I'm just giving it a little bit more structure.

Reply:
This is the original one https://github.com/DataDog/integrations-core/pull/13117/files#diff-bdee1557553b0662687ec083f65fd639f102682d8f8da248fa8744acbc34f7f5L207